services/jenkins/jenkins-tests.service.js

```diff
@@ -1,110 +1,154 @@
     return 'build'
   }
 
+  static get defaultBadgeData() {
+    return {
+      label: 'tests',
+    }
+  }
+
   static get route() {
     return {
-      base: 'jenkins/t',
-      pattern: ':scheme(http|https)?/:host/:job*',
+      base: 'jenkins/tests',
+      pattern: ':protocol(http|https)/:host/:job+',
+      queryParamSchema: queryParamSchema.concat(testResultQueryParamSchema),
     }
   }
 
   static get examples() {
     return [
       {
         title: 'Jenkins tests',
-        pattern: ':scheme/:host/:job',
         namedParams: {
-          scheme: 'https',
+          protocol: 'https',
           host: 'jenkins.qa.ubuntu.com',
           job:
-            'view/Precise/view/All%20Precise/job/precise-desktop-amd64_default',
+            'view/Trusty/view/Smoke%20Testing/job/trusty-touch-flo-smoke-daily',
         },
-        staticPreview: {
-          label: 'build',
-          message: 'passing',
-          color: 'brightgreen',
+        queryParams: {
+          compact_message: null,
+          passed_label: 'passed',
+          failed_label: 'failed',
+          skipped_label: 'skipped',
         },
+        staticPreview: this.render({
+          passed: 477,
+          failed: 2,
+          skipped: 0,
+          total: 479,
+          isCompact: false,
+        }),
+        documentation: getDocumentation({
+          route:
+            '/jenkins/tests/https/jenkins.qa.ubuntu.com/view/Trusty/view/Smoke%20Testing/job/trusty-touch-flo-smoke-daily',
+        }),
       },
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache }) {
-    camp.route(
-      /^\/jenkins(?:-ci)?\/t\/(http(?:s)?)\/([^/]+)\/(.+)\.(svg|png|gif|jpg|json)$/,
-      cache((data, match, sendBadge, request) => {
-        const scheme = match[1] // http(s)
-        const host = match[2] // example.org:8080
-        const job = match[3] // folder/job
-        const format = match[4]
-        const options = {
-          json: true,
-          uri: `${scheme}://${host}/job/${job}/lastBuild/api/json?tree=${encodeURIComponent(
-            'actions[failCount,skipCount,totalCount]'
-          )}`,
-        }
-        if (job.indexOf('/') > -1) {
-          options.uri = `${scheme}://${host}/${job}/lastBuild/api/json?tree=${encodeURIComponent(
-            'actions[failCount,skipCount,totalCount]'
-          )}`
-        }
+  static render({
+    passed,
+    failed,
+    skipped,
+    total,
+    passedLabel,
+    failedLabel,
+    skippedLabel,
+    isCompact,
+  }) {
+    return renderTestResultBadge({
+      passed,
+      failed,
+      skipped,
+      total,
+      passedLabel,
+      failedLabel,
+      skippedLabel,
+      isCompact,
+    })
+  }
 
-        if (serverSecrets.jenkins_user) {
-          options.auth = {
-            user: serverSecrets.jenkins_user,
-            pass: serverSecrets.jenkins_pass,
-          }
-        }
+  transform({ json }) {
+    const testsObject = json.actions.find(o => o.hasOwnProperty('failCount'))
```

Comment:
`.hasOwnProperty` is a pretty archaic form that I don't think we need to use anymore.

```suggestion
    const testsObject = json.actions.find(o => 'failCount' in o)
```